controller_manager:
  ros__parameters:
    update_rate: 30  # [Hz]

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    diffbot_base_controller:
      type: diff_drive_controller/DiffDriveController

#    vacuum_controller:
#      type: forward_command_controller/ForwardCommandController

    sweeper_controller:
      type: velocity_controllers/JointGroupVelocityController

    range_sensor_broadcaster_left:
      type: range_sensor_broadcaster/RangeSensorBroadcaster

    range_sensor_broadcaster_front:
      type: range_sensor_broadcaster/RangeSensorBroadcaster

    range_sensor_broadcaster_right:
      type: range_sensor_broadcaster/RangeSensorBroadcaster

    gpio_controller:
      type: vacuum_cleaner/GPIOController

diffbot_base_controller:
  ros__parameters:
    publish_rate: 30.0

    base_frame_id: base_link
    left_wheel_names: ['wheel_left_joint']
    right_wheel_names: ['wheel_right_joint']
    
    wheel_separation: 0.231                     # [m] - for rotation
    wheel_radius: 0.032                         # [m] - for speed
    wheels_per_side: 1

    linear.x.has_velocity_limits: true
    #linear.x.has_acceleration_limits: true
    linear.x.max_velocity: 0.10                 # [m/s] - v[m/s] = cmd_val[rad/s] * wheel_radius
    linear.x.min_velocity: -0.10                # [m/s]
    #linear.x.max_acceleration: 1.0              # [m/s²]

    angular.z.has_velocity_limits: true
    #angular.z.has_acceleration_limits: true
    angular.z.max_velocity: 0.8658008658        # [rad/s] - cmd_val[rad/s] = v[mm/s] / ( wheel_seperation[mm] / 2 )
    angular.z.min_velocity: -0.8658008658       # [rad/s]
    #angular.z.max_acceleration: 1.0             # [rad/s²]
    #angular.z.min_acceleration: -1.0            # [rad/s²]

    use_stamped_vel: false

#vacuum_controller:
#  ros__parameters:
#    joints:
#      - vacuum_joint
#    interface_name: PWM

sweeper_controller:               # ros2 topic pub --once /sweeper_controller/commands std_msgs/msg/Float64MultiArray "{data:[40.0,40.0]}"
  ros__parameters:
    joints:
      - sweeper_left_joint
      - sweeper_right_joint

range_sensor_broadcaster_left:
  ros__parameters:
    sensor_name: "range_sensor_left"
    frame_id: "range_sensor_left"
    radiation_type: 1
    field_of_view: 0.1
    min_range: 0.10
    max_range: 7.0

range_sensor_broadcaster_front:
  ros__parameters:
    sensor_name: "range_sensor_front"
    frame_id: "range_sensor_front"
    radiation_type: 1
    field_of_view: 0.1
    min_range: 0.10
    max_range: 7.0

range_sensor_broadcaster_right:
  ros__parameters:
    sensor_name: "range_sensor_right"
    frame_id: "range_sensor_right"
    radiation_type: 1
    field_of_view: 0.1
    min_range: 0.10
    max_range: 7.0

gpio_controller:
  ros__parameters:
    inputs:
      - user_interface/button_circle
      - user_interface/button_start
      - user_interface/button_home
      - limit_switch_left/state
      - limit_switch_right/state
    outputs:
      - user_interface/LED_circle
      - user_interface/LED_start_green
      - user_interface/LED_start_red
      - user_interface/LED_home